### DISCUSSION TITLE
Converting existing project?

### DISCUSSION THREAD

**Discussion by khuttun** [BrainiumLLC/cargo-mobile#15] [Q&A]
What would be a good way to convert an existing wgpu desktop project to use cargo-mobile? Have you planned any automation for this use case?

**Comment by francesca64** [comment]
Sorry for the delay, and thanks for asking!

There's currently no super nice way to do this. The process for doing it manually should look something like this:

1. In another directory, generate a wgpu project using cargo-mobile (or any other type of project - it shouldn't matter)
2. Copy the generated `mobile.toml`, `gen/`, and `.cargo/` into your existing project
3. Copy the generated `.gitignore`'s section titled `# cargo-mobile` into your existing project's `.gitignore`
4. Copy the generated `Cargo.toml`'s `[lib]`, `[[bin]]`, and `[target.'cfg(target_os = "android")'.dependencies]` sections along with the `mobile-entry-point` dependency into your existing `Cargo.toml`
5. Rename your existing `main.rs` to `lib.rs`
6. Annotate your `main` function with `#[mobile_entry_point]` (with a `use mobile_entry_point::mobile_entry_point;` somewhere)

Automating this would certainly be a huge improvement. Steps 1 and 2 can be eliminated completely if we make it possible to use an existing project directly. Steps 3 and 5 are easy to automate. Steps 4 and 6 could be impractical to automate _completely_ correctly, but a best-effort implementation would likely be adequate for most people.

**Reply by khuttun** [reply]
Thanks, this is pretty much exactly what I ended up doing. Actually automating 1 & 2 could be the first step towards direction where cargo-mobile could be used with a Rust project without storing the Android Studio/Xcode files with the project's code. cargo-mobile could just generate those platform specific files when needed. What do you think about this kind of direction for cargo-mobile?
